core/src/main/java/org/apache/calcite/sql/dialect/StarRocksSqlDialect.java

```diff
@@ -93,6 +94,18 @@ public StarRocksSqlDialect(Context context) {
     return true;
   }
 
+  @Override public boolean supportsUnparseOnlyDeterministic(SqlOperator sqlOperator) {
```

Comment:
I am not sure this function is useful.
If the function returns 'false' and you cannot unparse, you will need to throw.
If you remove the function, you still throw in the same places.

Reply:
If we do not support rewriting this function, we should throw exceptions at the point of rewriting. We directly convert it because only Starrocks has been identified to have this issue. If it is discovered later that operations like A IS TRUE and A IS NOT NULL are also unsupported, exceptions must be thrown during the conversion process. This conversion logic can be implemented as part of the Dialect behavior.
If I remove this function, the A IS FALSE will remain unchanged and not be converted. 
What I'm thinking here is that not all dialects have undergone such transformation. If the database supports A IS TRUE, then we won't make any changes.